src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -848,176 +848,124 @@ public ListenableFuture<DescribeIndexStatsResponse> describeIndexStats(Struct fi
-     *  <pre>{@code
-     *     import io.pinecone.proto.ListResponse;
-     *     import com.google.common.util.concurrent.Futures;
-     *     import com.google.common.util.concurrent.ListenableFuture;
-     *
-     *     ...
-     *
-     *     ListenableFuture<ListResponse> futureResponse = asyncIndex.list("example-namespace", "prefix-", "some-pagToken", 10);
-     *     ListResponse asyncListResponse = Futures.getUnchecked(futureResponse);
-     *  }</pre>
-     *
-     * @param namespace The namespace that holds the vector IDs you want to retrieve. If namespace is not specified,
-     *      *                  the default namespace is used.
-     * @param prefix The prefix with which vector IDs must start to be included in the response.
-     * @param paginationToken The token to paginate through the list of vector IDs.
-     * @param limit The maximum number of vector IDs you want to retrieve.
-     * @return {@link ListResponse} containing the list of vector IDs fetched from the specified namespace.
-     *         The response includes vector IDs up to {@code 100} items.
-     * @throws RuntimeException if there are issues processing the request or communicating with the server.
-     *         This includes network issues, server errors, or serialization issues with the request or response.
-     */
-    @Override
-    public ListenableFuture<ListResponse> list(String namespace, String prefix, String paginationToken, int limit) {
-        validateListEndpointParameters(namespace, prefix, paginationToken, limit, true, true, true, true);
-        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).
-                setPaginationToken(paginationToken).setLimit(limit).build();
-        return asyncStub.list(listRequest);
+        /**
+         * Sets the pagination token to retrieve the next page of vector IDs.
+         * @param paginationToken The pagination token to retrieve the next page of vector IDs.
+         * @return The {@link AsyncIndex.ListRequestBuilder} object.
+         */
+        public ListRequestBuilder setPaginationToken(String paginationToken) {
+            this.paginationToken = paginationToken;
+            return this;
+        }
+
+        /**
+         * Sets the limit of vector IDs to retrieve.
+         * @param limit The limit of vector IDs to retrieve.
+         * @return The {@link AsyncIndex.ListRequestBuilder} object.
+         */
+        public ListRequestBuilder setLimit(Integer limit) {
+            this.limit = limit;
+            return this;
+        }
+
+        /**
+         * Builds the {@link ListRequest} object to send to the list endpoint in order to retrieve vector IDs back from an index.
+         * @return The {@link ListenableFuture} object.
+         */
+        public ListenableFuture<ListResponse> build() {
+            // Logic to validate parameters
+            validateListEndpointParameters(namespace, prefix, paginationToken, limit,
```

Comment:
To be honest I'm still really confused about all the booleans in this validation function (`namespaceRequired`, `prefixRequired`, `paginationTokenRequest`, `limitRequired`). I'm just really not clear why we need to do validations this way. 

For example, I don't think we need to validate the namespace at all right? They can pass either "" for default if they want or nothing. Am I misunderstanding an issue here?